parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -272,6 +279,23 @@ pub mod pallet {
 			let enqueued_message: EnqueuedMessage =
 				EnqueuedMessage::decode(&mut message).map_err(|_| ProcessMessageError::Corrupt)?;
 
+			// Skp halt check for Upgrade and SetOperatingMode
+			let halt_check_required = !matches!(
```

Comment:
We run into the same problem where we need to decode all the pending messages before deciding whether to yield or not.

I want you to try and find a design that relies solely on the origin passed to process_message to decide whether to yield.

Consider using an origin type like this:
Struct 

```rust 
struct AggregateMessageOrigin {
  Self(Priority)
  Parachain(ParaId)
}
```

Then in `process_message`, if origin != Self(Priority::High), when we should yield the message.

Reply:
Cool! So does that implicitly mean you prefer  to give `create_agent`  a low Priority to address my previous concern in https://github.com/Snowfork/snowbridge/pull/949#discussion_r1357700866?

> specifically assuming message queue for bridge hub has already been spammed by the `create_agent` message, then the upgrade will never get the chance to execute.

Reply:
Yes, let's give create-agent a lower priority.

Reply:
But I don't want the priority to be a parameter passed to OutboundQueue::submit.

Instead OutboundQueue should just check if the command is an upgrade or operating mode change and then prioritise accordingly

Reply:
Yes, that's exactly what the current implementation does. We check priority in `validate` phase and set it to the `ticket` accordingly.